packages/nvidia_nat_mcp/src/nat/plugins/mcp/client_impl.py

```
@@ -250,13 +432,17 @@ async def mcp_client_function_group(config: MCPClientConfig, _builder: Builder):
             function_name = override.alias if override and override.alias else tool_name
             description = override.description if override and override.description else tool.description
 
-            # Create the tool function
-            tool_fn = mcp_tool_function(tool)
+            # Create the tool function according to configuration
+            if config.session_aware_tools:
+                tool_fn = mcp_session_tool_function(tool, group)
+            else:
+                from nat.plugins.mcp.tool import mcp_tool_function  # noqa: F401
```

Ruff (0.13.2)

Unused `noqa` directive (non-enabled: `F401`)

Remove unused `noqa` directive

(RUF100)